api/src/main/java/com/epam/pipeline/controller/metadata/CategoricalAttributeController.java

```diff
@@ -0,0 +1,126 @@
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.util.List;
+import java.util.Map;
+
+@Controller
+@Api(value = "CategoricalAttributes")
+@RequiredArgsConstructor
+@RequestMapping(value = "/categoricalAttribute")
+public class CategoricalAttributeController extends AbstractRestController {
+
+    private final CategoricalAttributeApiService categoricalAttributeApiService;
+
+    @PostMapping
+    @ResponseBody
+    @ApiOperation(
+        value = "Add categorical attributes values.",
+        notes = "Add categorical attributes values",
+        produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiResponses(
+        value = {@ApiResponse(code = HTTP_STATUS_OK, message = API_STATUS_DESCRIPTION)
+        })
+    public Result<Boolean> insertCategoricalAttributes(@RequestBody final Map<String, List<String>> dict) {
+        return Result.success(categoricalAttributeApiService.insertAttributesValues(dict));
+    }
+
+    @GetMapping("/loadAll")
+    @ResponseBody
+    @ApiOperation(
+        value = "Load all categorical attributes with values.",
+        notes = "Load all categorical attributes with values.",
+        produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiResponses(
+        value = {@ApiResponse(code = HTTP_STATUS_OK, message = API_STATUS_DESCRIPTION)
+        })
+    public Result<Map<String, List<String>>> loadAllCategoricalAttributes() {
+        return Result.success(categoricalAttributeApiService.loadAll());
+    }
+
+    @GetMapping("/load")
+    @ResponseBody
+    @ApiOperation(
+        value = "Load all requested attributes with values.",
+        notes = "Load all requested attributes with values.",
+        produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiResponses(
+        value = {@ApiResponse(code = HTTP_STATUS_OK, message = API_STATUS_DESCRIPTION)
+        })
+    public Result<Map<String, List<String>>> loadCategoricalAttributes(@RequestParam final List<String> attributeKeys) {
+        return Result.success(categoricalAttributeApiService.loadAllValuesForKeys(attributeKeys));
+    }
+
+    @DeleteMapping("/delete")
+    @ResponseBody
+    @ApiOperation(
+        value = "Delete all values for a requested attribute.",
+        notes = "Delete all values for a requested attribute.",
+        produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiResponses(
+        value = {@ApiResponse(code = HTTP_STATUS_OK, message = API_STATUS_DESCRIPTION)
+        })
+    public Result<Boolean> deleteAllAttributeValues(@RequestParam final String attributeKey) {
+        return Result.success(categoricalAttributeApiService.deleteAttributeValues(attributeKey));
+    }
+
+    @DeleteMapping("/deleteValue")
```

Comment:
Also, let's do not action to path and use `params` in `DeleteMapping` annotation

Reply:
Done